### PR TITLE
Search repositories

### PR DESCRIPTION
**Pull request by leichunfeng**

### CODE REVIEW

File: OctoKit/OCTRepositoriesSearchResult.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@interface OCTRepositoriesSearchResult : OCTObject
+
+// The total repositories count of the search results.
+@property (nonatomic, assign, readonly) NSUInteger totalCount;
+
+// Indicates whether the results incomplete or not.
+@property (nonatomic, assign, getter = isIncompleteResults, readonly) BOOL incompleteResults;
+
+// The repository array of the search results.
+@property (nonatomic, strong, readonly) NSArray *repositories;
```
Comment (joshaber) [inline OctoKit/OCTRepositoriesSearchResult.h]: Style: let's make this `copy` instead.

File: OctoKit/OCTClient+Search.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#import <OctoKit/OctoKit.h>
+
+@interface OCTClient (Search)
+
+///  Search repositories.
+///
+///  query - The search keywords, as well as any qualifiers. This must not be nil.
+///  sort  - The sort field. One of stars, forks, or updated. Default: results
+///          are sorted by best match. This can be nil.
+///  order - The sort order if sort parameter is provided. One of asc or desc.
```
Comment (joshaber) [inline OctoKit/OCTClient+Search.h]: Could this be an enum since it's just two fixed choices?

**Comment by joshaber** [comment]
Awesome, thanks so much! Just a couple questions.

**Comment by leichunfeng** [comment]
I have finished it, please review it again.

File: OctoKit/OCTClient+Search.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+///  Search repositories.
+///
+///  query   - The search keywords, as well as any qualifiers. This must not be nil.
+///  orderBy - The sort field. One of stars, forks, or updated. Default: results
+///            are sorted by best match. This can be nil.
+///  asc     - The sort order, asc or not.
+///
+///  Returns a signal which will send the 
search result `OCTRepositoriesSearchResult`.
+- (RACSignal *)searchRepositoriesWithQuery:(NSString *)query orderBy:(NSString *)orderBy asc:(BOOL)asc;
```
Comment (joshaber) [inline OctoKit/OCTClient+Search.h]: Let's spell out `ascending`.

**Comment by joshaber** [comment]
:metal: Just one nit pick left.

**Comment by leichunfeng** [comment]
OK, I have finished it.

**Comment by joshaber** [comment]
:boom: Thanks!

**Pull request merged**